### PR TITLE
Merge master HEAD into openj9-staging

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Merge master HEAD into openj9-staging

Conflicts:
```
	make/modules/java.desktop/lib/ClientLibraries.gmk
	make/modules/jdk.jcmd/Java.gmk
	src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java
```

Requires
* https://github.com/eclipse-openj9/openj9/pull/19335

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: make/modules/java.desktop/lib/ClientLibraries.gmk [hunk 1]
```diff
@@ -412,14 +293,6 @@ else
      HARFBUZZ_CFLAGS += -DHAVE_INTEL_ATOMIC_PRIMITIVES -DHB_NO_VISIBILITY
    endif
 
-   # Early re-canonizing has to be disabled to workaround an internal XlC compiler error
-   # when building libharfbuzz
-   ifeq ($(call isTargetOs, aix), true)
-    ifneq ($(TOOLCHAIN_TYPE), clang)
-     HARFBUZZ_CFLAGS += -qdebug=necan
-    endif
-   endif
```
Comment (keithc-ca) [inline make/modules/java.desktop/lib/ClientLibraries.gmk:421]: This needs to remain (in `ClientLibraries.gmk`) for AIX builds with xlc.

File: src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java [hunk 2]
```diff
@@ -39,26 +34,23 @@ class ClassLoaderHelper {
...
+     * This method simply returns null.  It could implement the alternate name
+     * converting ".so" with ".a" suffix but redundant.
      */
     static File mapAlternativeName(File lib) {
-        String name = lib.toString();
-        int index = name.lastIndexOf('.');
-        if (index < 0) {
-            return null;
-        }
-        return new File(name.substring(0, index) + ".a");
```
Comment (keithc-ca) [inline src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java:61]: We need to keep this unless `JVM_LoadLibrary()` in OpenJ9 is changed to handle loading from `.a` files like the referenced implementation. I think the rest of this class can be the same as upstream.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: make/modules/java.desktop/lib/ClientLibraries.gmk (hunk 1 above)
Comment (JasonFengJ9) [reply make/modules/java.desktop/lib/ClientLibraries.gmk]: Restored

File: src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java (hunk 2 above)
Comment (JasonFengJ9) [reply src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java]: Added it back.

Comment (pshipton) [reply src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java]: We should match the RI, although it doesn't need to happen in this PR.
Created https://github.com/eclipse-openj9/openj9/issues/19344

File: make/modules/java.desktop/lib/ClientLibraries.gmk [hunk 3]
```diff
@@ -23,16 +23,11 @@
...
-
-LIBAWT_DEFAULT_HEADER_DIRS := \
-    common/awt/utility \
-    libawt/awt/image \
-    libawt/awt/image/cvutils \
-    libawt/java2d \
-    libawt/java2d/loops \
-    libawt/java2d/pipe \
-    #
+################################################################################
```
Comment (keithc-ca) [inline make/modules/java.desktop/lib/ClientLibraries.gmk:26]: This file needs an IBM copyright notice.

File: src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java [hunk 4]
```diff
@@ -22,10 +22,9 @@
  * or visit www.oracle.com if you need additional information or have any
  * questions.
  */
-
```
Comment (keithc-ca) [inline src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java:25]: Please keep this blank line (consistent with other `.java` source files).

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: make/modules/java.desktop/lib/ClientLibraries.gmk (hunk 3 above)
Comment (JasonFengJ9) [reply make/modules/java.desktop/lib/ClientLibraries.gmk:32]: Added

File: src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java (hunk 4 above)
Comment (JasonFengJ9) [reply src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java:25]: Restored it.

File: make/modules/java.desktop/lib/ClientLibraries.gmk [hunk 5]
```diff
@@ -23,16 +23,17 @@
 # questions.
 #
 
-include CopyFiles.gmk
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2024, 2024 All Rights Reserved
+ * ===========================================================================
+ */
```
Comment (keithc-ca) [inline make/modules/java.desktop/lib/ClientLibraries.gmk:30]: Comments begin with `#` in makefiles; we want this immediately after line 24:
```
# ===========================================================================
# (c) Copyright IBM Corp. 2024, 2024 All Rights Reserved
# ===========================================================================
```

File: src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java [hunk 6]
```diff
@@ -39,18 +39,20 @@ class ClassLoaderHelper {
...
     /**
-     * Returns an alternate path name for the given file
-     * such that if the original pathname did not exist, then the
-     * file may be located at the alternate location.
-     * For AIX, this replaces the final .so suffix with .a
+     * AIX implementation of JVM_LoadLibrary handles the alternate path name mapping.
+     * If loading of the given library name with ".so" suffix fails, it will attempt
+     * to load the library of the same name with ".a" suffix as the alternate name.
+     * This method simply returns null.  It could implement the alternate name
+     * converting ".so" with ".a" suffix but redundant.
```
Comment (keithc-ca) [inline src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java:55]: Please don't change this comment; the one from upstream is incorrect for OpenJ9.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: make/modules/java.desktop/lib/ClientLibraries.gmk (hunk 5 above)
Comment (JasonFengJ9) [reply make/modules/java.desktop/lib/ClientLibraries.gmk]: Fixed the comments.

File: src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java (hunk 6 above)
Comment (JasonFengJ9) [reply src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java]: Restore the comment.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**